tests/foreman/cli/test_subnet.py

```diff
@@ -308,25 +290,20 @@ def test_negative_update_2(self, options):
 
         """
         subnet = make_subnet()
-        opts = {u'id': subnet['id']}
-        # generate pool range from network address
-        for key, val in options.iteritems():
-            opts[key] = re.sub(r'\d+$', str(val), subnet['network'])
-        with self.assertRaises(CLIReturnCodeError):
-            Subnet.update(opts)
-        # check - subnet is not updated
-        result = Subnet.info({u'id': subnet['id']})
-        for key in options.keys():
-            self.assertEqual(result[key], subnet[key])
-
-    @data(
-        gen_string(str_type='alpha'),
-        gen_string(str_type='alphanumeric'),
-        gen_string(str_type='numeric'),
-        gen_string(str_type='latin1'),
-        gen_string(str_type='utf8'),
-    )
-    def test_positive_delete_1(self, test_name):
+        for options in invalid_addr_pools():
+            with self.subTest(options):
+                opts = {u'id': subnet['id']}
+                # generate pool range from network address
+                for key, val in options.iteritems():
```

Comment:
not sure whether that statement will work properly (but it is more lack of knowledge from my side)